src/main/java/com/amazonaws/encryptionsdk/keyrings/MultiKeyring.java

```diff
@@ -83,11 +87,11 @@ public void onDecrypt(DecryptionMaterials decryptionMaterials, List<? extends En
 
         for (Keyring keyring : keyringsToDecryptWith) {
             try {
-                keyring.onDecrypt(decryptionMaterials, encryptedDataKeys);
+                final DecryptionMaterials resultMaterials = keyring.onDecrypt(decryptionMaterials, encryptedDataKeys);
```

Comment:
Same comment as with `onEncrypt`

Reply:
OnDecrypt catches all exceptions and only fails if every keyring failed.